lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -0,0 +1,337 @@
+  #toggleEnableTimers(activate) {
+    const options = {
+      toFake: {
+        setTimeout: () => {
+          this.#realSetTimeout = globalThis.setTimeout;
+          this.#realClearTimeout = globalThis.clearTimeout;
+          this.#realTimersSetTimeout = nodeTimers.setTimeout;
+          this.#realTimersClearTimeout = nodeTimers.clearTimeout;
+          this.#realPromisifiedSetTimeout = nodeTimersPromises.setTimeout;
+
+          globalThis.setTimeout = this.#setTimeout;
+          globalThis.clearTimeout = this.#clearTimeout;
+
+          nodeTimers.setTimeout = this.#setTimeout;
+          nodeTimers.clearTimeout = this.#clearTimeout;
+
+          nodeTimersPromises.setTimeout = FunctionPrototypeBind(
+            this.#setTimeoutPromisified,
+            this,
+          );
+        },
+        setInterval: () => {
+          this.#realSetInterval = globalThis.setInterval;
+          this.#realClearInterval = globalThis.clearInterval;
+          this.#realTimersSetInterval = nodeTimers.setInterval;
+          this.#realTimersClearInterval = nodeTimers.clearInterval;
+          this.#realPromisifiedSetInterval = nodeTimersPromises.setInterval;
+
+          globalThis.setInterval = this.#setInterval;
+          globalThis.clearInterval = this.#clearInterval;
+
+          nodeTimers.setInterval = this.#setInterval;
+          nodeTimers.clearInterval = this.#clearInterval;
+
+          nodeTimersPromises.setInterval = FunctionPrototypeBind(
+            this.#setIntervalPromisified,
+            this,
+          );
+        },
+      },
+      toReal: {
+        setTimeout: () => {
+          globalThis.setTimeout = this.#realSetTimeout;
+          globalThis.clearTimeout = this.#realClearTimeout;
+
+          nodeTimers.setTimeout = this.#realTimersSetTimeout;
+          nodeTimers.clearTimeout = this.#realTimersClearTimeout;
+
+          nodeTimersPromises.setTimeout = this.#realPromisifiedSetTimeout;
+        },
+        setInterval: () => {
+          globalThis.setInterval = this.#realSetInterval;
+          globalThis.clearInterval = this.#realClearInterval;
+
+          nodeTimers.setInterval = this.#realTimersSetInterval;
+          nodeTimers.clearInterval = this.#realTimersClearInterval;
+
+          nodeTimersPromises.setInterval = this.#realPromisifiedSetInterval;
+        },
+      },
+    };
+
+    const target = activate ? options.toFake : options.toReal;
+    ArrayPrototypeForEach(this.#timersInContext, (timer) => target[timer]());
+    this.#isEnabled = activate;
+  }
+
+  tick(time = 1) {
+    if (!this.#isEnabled) {
+      throw new ERR_INVALID_STATE(
+        'You should enable MockTimers first by calling the .enable function',
+      );
+    }
+
+    if (time < 0) {
+      throw new ERR_INVALID_ARG_TYPE(
+        'time',
+        'positive integer',
+        time,
+      );
+    }
+
+    this.#now += time;
+    let timer = this.#executionQueue.peek();
+    while (timer) {
+      if (!(timer.runAt <= this.#now)) break;
```

Comment:
Could `!(timer.runAt <= this.#now)` be simplified to `timer.runAt > this.#now`?

Reply:
thanks!